src/control/Control.cpp

```diff
@@ -1235,6 +1238,48 @@ void Control::deletePage() {
 
 void Control::insertNewPage(size_t position) { pageBackgroundChangeController->insertNewPage(position); }
 
+void Control::appendNewPdfPages() {
+    int pageCount = this->doc->getPageCount();
+    // find last page with pdf background and get its pdf page number
+    PageRef page;
+    int lastPdfPage = -1;
+    for (int i = pageCount - 1; i >= 0; --i) {
+        page = doc->getPage(i);
+        if (page && page->getBackgroundType().isPdfPage()) {
+            lastPdfPage = page->getPdfPageNr();
+            break;
+        }
+    }
+    if (lastPdfPage == -1) {
+        string msg = FS(_F("Document has no pdf pages."));
+        XojMsgBox::showErrorToUser(getGtkWindow(), msg);
+        return;
+    }
+
+    int pdfPageCount = this->doc->getPdfPageCount();
+    int numPages = pdfPageCount - lastPdfPage - 1;
+
+    if (numPages <= 0) {
```

Comment:
Also use auto or size_t.
```suggestion
    auto pdfPageCount = this->doc->getPdfPageCount();
    auto insertCount = pdfPageCount - currentPdfPageCount;

    if (insertCount == 0) {
```